src/infra/core/src/utils/datasets_filtering.rs

```diff
@@ -41,10 +63,146 @@ pub fn filter_datasets_by_pattern(
                 future::ready(
                     dataset_ref_patterns
                         .iter()
-                        .any(|dataset_ref_pattern| dataset_ref_pattern.is_match(dataset_handle)),
+                        .any(|dataset_ref_pattern| dataset_ref_pattern.matches(dataset_handle)),
                 )
             })
             .map_err(Into::into)
             .boxed()
     }
 }
+
+////////////////////////////////////////////////////////////////////////////////
+
+pub fn filter_datasets_by_any_pattern(
+    dataset_repo: &dyn DatasetRepository,
+    search_svc: Arc<dyn SearchService>,
+    dataset_ref_any_patterns: Vec<DatasetRefAnyPattern>,
+) -> FilteredDatasetRefAnyStream<'_> {
+    let in_multitenant_mode = dataset_repo.is_multi_tenant();
+
+    let (all_ref_patterns, static_refs): (Vec<_>, Vec<_>) = dataset_ref_any_patterns
+        .into_iter()
+        .partition(DatasetRefAnyPattern::is_pattern);
+
+    let (remote_ref_patterns, local_ref_patterns): (Vec<_>, Vec<_>) = all_ref_patterns
+        .into_iter()
+        .partition(|pattern| pattern.is_remote_pattern(in_multitenant_mode));
+
+    let static_datasets_stream = get_static_datasets_stream(static_refs);
+    let remote_patterns_stream =
+        get_remote_datasets_stream(search_svc, remote_ref_patterns, in_multitenant_mode);
+    let local_patterns_stream = get_local_datasets_stream(dataset_repo, local_ref_patterns);
+
+    static_datasets_stream
+        .chain(remote_patterns_stream)
+        .chain(local_patterns_stream)
+        .boxed()
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+fn get_static_datasets_stream(
+    static_refs: Vec<DatasetRefAnyPattern>,
+) -> FilteredDatasetRefAnyStream<'static> {
+    Box::pin(async_stream::try_stream! {
+        for static_ref in &static_refs.clone() {
```

Comment:
Unnecessary clone - we already own the `Vec` here.

Also if we use `static_refs.into_iter()` the below
```rust
yield static_ref
                .as_dataset_ref_any()
                .unwrap()
                .clone();
```
could be made into
```rust
yield static_ref
                .into_dataset_ref_any()
                .unwrap();
```
to avoid all the cloning.